### PR TITLE
Feature/libbfio read

### PR DESCRIPTION
**Pull request by omerbenamram**

### CODE REVIEW

File: libbfio/src/io_handle.rs [hunk 1]
```diff
@@ -1,156 +1,171 @@
...
-    }
+    trace!("io_handle_seek");
 
     let seek_from = match whence as u32 {
         SEEK_SET => SeekFrom::Start(offset),
         SEEK_END => SeekFrom::End(offset as i64),
         SEEK_CUR => SeekFrom::Current(offset as i64),
-        _ => panic!("unexpected `whence`"),
+        _ => {
+            libcerror_error_set(
```
Comment (ohadravid) [inline libbfio/src/io_handle.rs:125]: I think you could put a nice
```
enum ErrorCode {
...
}

pub fn set_error(error: ..., error_code: ErrorCode, format: &str, msg: &str) {
}
```

in the `libcerror_sys` crate

File: libcerror-sys/build.rs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        PathBuf::from(local_install)
+    } else {
+        env::current_dir().unwrap().join("libcerror")
+    };
+
+    if cfg!(target_os = "windows") {
+        println!("cargo:rustc-link-lib=static=libcerror");
+
+        // Also static-link deps (otherwise we'll get missing symbols at link time).
+        println!("cargo:rustc-link-lib=static=libcerror");
```
Comment (ohadravid) [inline libcerror-sys/build.rs]: Does it really depend on anything?

File: libbfio/src/handle.rs [hunk 3]
```diff
@@ -45,68 +48,119 @@ impl Handle {
     }
 }
 
-type io_handle_t = *mut IoHandle;
+pub type IoHandleConstRef = *const IoHandle;
+pub type IoHandleRefMut = *mut IoHandle;
+pub type BoxedIoHandleRefMut = *mut *mut IoHandle;
```
Comment (ohadravid) [inline libbfio/src/handle.rs]: As I told you, I don't really like those names.
Also, why is `*mut *mut` a `Boxed`? You can do a `&&File::new() as *mut *mut` or something..

File: libbfio/src/handle.rs [hunk 4]
```diff
@@ -45,68 +48,119 @@ impl Handle {
...
+//    CloneByReference = 0x02,
+//}
+
+/* The access flags definitions
+ * bit 1						set to 1 for read access
+ * bit 2						set to 1 for write access
+ * bit 3						set to 1 to truncate an existing file on write
+ * bit 4-8						not used
+ */
+pub enum LibbfioAccessFlags {
```
Comment (ohadravid) [inline libbfio/src/handle.rs:75]: Nice. Doesn't bindgen generate anything that you can use for the value part?

File: libbfio/src/handle.rs [hunk 5]
```diff
@@ -201,53 +255,51 @@ extern "C" {
...
                 Some(io_handle_read),
                 Some(io_handle_write),
                 Some(io_handle_seek),
-                Some(io_handle_exists),
+                None,
                 Some(io_handle_is_open),
                 Some(io_handle_get_size),
-                0,
+                // LibbfioFlagIoHandleManaged
+                1,
```
Comment (ohadravid) [inline libbfio/src/handle.rs]: Extract enum?

File: libbfio/src/handle.rs [hunk 6]
```diff
@@ -201,53 +255,51 @@ extern "C" {
...
-
-        unsafe {
-            libbfio_handle_free(&mut self.as_type_ref_mut() as *mut _, &mut error);
-        }
-
-        debug_assert!(error.is_null(), "`{}` failed!", module_path!());
-    }
-}
-
 impl Handle {
```
Comment (ohadravid) [inline libbfio/src/handle.rs:247]: Document what this is? (aka vs IoHandle)

**Review by ohadravid — CHANGES_REQUESTED** [review]

File: libcerror-sys/build.rs (hunk 2 above)
Comment (omerbenamram) [reply libcerror-sys/build.rs]: on itself

File: libbfio/src/handle.rs (hunk 4 above)
Comment (omerbenamram) [reply libbfio/src/handle.rs:75]: it was a bit annoying, but I've used the imported types

File: libbfio/src/handle.rs (hunk 5 above)
Comment (omerbenamram) [reply libbfio/src/handle.rs]: done

File: libbfio/src/handle.rs (hunk 3 above)
Comment (omerbenamram) [reply libbfio/src/handle.rs]: I've removed them, but I still cant use the bindgen imports unless I'll cast everything to opaque pointer types.

File: libbfio/src/handle.rs (hunk 6 above)
Comment (omerbenamram) [reply libbfio/src/handle.rs:247]: I added some module-level docs

**Review by ohadravid — APPROVED** [review]

**Pull request merged**